source/agora/common/Serializer.d

```diff
@@ -407,22 +407,19 @@ public void serializePart (T) (scope const auto ref T record, scope SerializeDg
                                CompactMode compact = CompactMode.Yes)
     @safe
 {
-    import geod24.bitblob;
-
     // Custom serialization handling trumps everything else
     static if (hasSerializeMethod!T)
         record.serialize(dg);
 
-    // BitBlob are fixed size and thus, value types
-    // If we use an `ubyte[]` overload, the length gets serialized
-    else static if (is(T : BitBlob!N, size_t N))
-        dg(record[]);
-
     // Static array needs to be handled before arrays (because they convert)
     else static if (is(T : E[N], E, size_t N))
     {
-        foreach (ref elem; record)
-            serializePart(elem, dg, compact);
+        // Small type optimization
+        static if (!hasSerializeMethod!E && (E.sizeof == 1 || isSomeChar!E))
```

Comment:
`isSomeChar` will also return true for `dchar` which are 4 bytes, is this desired? And if it is, then why not write `E.sizeof ==4`?

Reply:
Returns true for `char`, `wchar`, `dchar`, and their qualified version.
Chars are a bit special, because they require validation. Casting straight from `ubyte[]` to `Char[]` is an error, even when `is(Char == dchar)` because the value could be outside the range of valid `dchar`.
But still, we don't want to serialize the `char` element piece-meal.

Looking at this from another angle, why can't we just do this optimization for any static array that doesn't have an element which defines `serialize` ? Well because this only checks one level down. Which makes me think, this approach might be a bit too simple...

Reply:
Yeah I don't think it's worth it (ATM) to check for nested static array, so let's keep it simple.